Worldview caching name problems

furlong-cmu opened the pull request:
I was having a problem opening worldview zip files that were not named with a particular naming format. 
This was causing a problem during data caching.

Changed worldview.py to query the zip file for a *.tif, and use that to extract the meta data about the sensor and collection date.

bcoltin commented on delta/imagery/sources/worldview.py:
```diff
@@ -101,10 +102,21 @@ def _prep(self, paths):
         assert isinstance(paths, str)
-        parts = os.path.basename(paths).split('_')
+        (_, ext) = os.path.splitext(paths)
+        if ext == '.zip':
```
Will this work with non-zip files? when we call unpack later I thought we assumed it was a zip. (not sure this check is needed)

furlong-cmu commented:
Updated the branch.  Removed the check and made it an assert, will fail if
the extension doesn't contain ".zip". This may be bad if we let people name
zip files without a .zip extension.
Also restored executable permission to model2config.

On Tue, Aug 18, 2020 at 11:38 PM Brian Coltin <notifications@github.com>
wrote:

> *@bcoltin* requested changes on this pull request.
>
> I also think you didn't mean to make scripts/model2config non-executable.
> ------------------------------
>
> In delta/imagery/sources/worldview.py
> <https://github.com/nasa/delta/pull/23#discussion_r472550241>:
>
> > @@ -101,10 +102,21 @@ def _prep(self, paths):
>             TODO: Apply TOA conversion!
>          """
>          assert isinstance(paths, str)
> -        parts = os.path.basename(paths).split('_')
> +        (_, ext) = os.path.splitext(paths)
> +        if ext == '.zip':
>
> Will this work with non-zip files? when we call unpack later I thought we
> assumed it was a zip. (not sure this check is needed)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nasa/delta/pull/23#pullrequestreview-469898413>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AALSQ7HAXLWBCVWB53XZEGDSBMGGHANCNFSM4QEH4YXQ>
> .
>
